src/App.js

```diff
@@ -1,25 +1,96 @@
-import logo from './logo.svg';
 import './App.css';
+import React, { useState, useEffect } from 'react';
+export default App;
 
 function App() {
+
+  const initialValues = { username: "", email: "", password: "" };
+  const [formValues, setFormValues] = useState(initialValues);
+  const [formErrors, setFormErrors] = useState({});
+  const [isSubmit, setIsSubmit] = useState(false);
+  const [isHovered, setIsHovered] = useState(false);
+
+  const handleMouseEnter = () => {
+    setIsHovered(true);
+  };
+
+  const handleMouseLeave = () => {
+    setIsHovered(false);
+  };
+  const buttonClass = isHovered ? 'hovered' : '';
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormValues({ ...formValues, [name]: value });
+  };
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setFormErrors(validate(formValues));
+    setIsSubmit(true);
+  };
+
+
+  useEffect(() => {
+    console.log(formErrors);
+    if (Object.keys(formErrors).length === 0 && isSubmit)
+      console.log(formValues);
+  }, [formErrors, formValues, isSubmit])
+
+
+  const validate = (values) => {
+    const errors = {}
+    const regex = /^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/i;
+
+    if (!values.username) {
+      errors.username = "Username is required!";
+    }
+    if (!values.email) {
+      errors.email = "Email is required!";
+    } else if (!regex.test(values.email))
+      errors.email = "This is not a valid email format!";
+
+    if (!values.password) {
+      errors.password = "Password is required!";
+    } else if (values.password < 5) {
+      errors.passowrd = "Passowrd must be more than 5 characters";
+    }
+    return errors;
+  };
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
-}
+    <React.Fragment>
+      <div className='wrapper'>
+        {/* <pre>{JSON.stringify(formValues, undefined, 2)}</pre> */}
+        <form onSubmit={handleSubmit} className="form">
+          <h1>Your form</h1>
+          <label>
+            Username
+            <br></br>
```

Comment:
'br' is an empty line tag and does not close itself. 
also, br is used to insert empty line IN TEXT. for example like <i></i>, <strong></strong> and other that work best IN TEXT. br is not a way of moving things to another line or to another position when it's not text. semantics is important (meaning what is what in html and using it accordingly). 
https://www.w3schools.com/tags/tag_br.asp